packages/block-editor/src/store/selectors.js

```diff
@@ -1798,3 +1798,18 @@ export function isBlockHighlighted( state, clientId ) {
 export function areInnerBlocksControlled( state, clientId ) {
 	return !! state.blocks.controlledInnerBlocks[ clientId ];
 }
+
+export function getHoveredBlocks( state ) {
+	return state.hoveredBlocks.hoveredBlockIds;
+}
+
+export function getHoveredBlockByBlockName( state, blockName ) {
+	const matchingBlockClientId = state.hoveredBlocks.hoveredBlockIds.find(
+		( blockId ) => getBlockName( state, blockId ) === blockName
+	);
+	return getBlock( state, matchingBlockClientId );
```

Comment:
I think it would be good to memoize this selector (maybe on `state.hoveredBlocks.hoveredBlockIds` and `state.blocks.cache`)

Reply:
✅  set up the memoize with `createSelector`